octoprint_mrbeam/software_update_information.py

```diff
@@ -98,233 +228,190 @@ def switch_software_channel(plugin, channel):
-        pip_command = GLOBAL_PIP_COMMAND
-    package_name = "netconnectd"
-    # get_package_description does not search for package version.
-    version = get_version_of_pip_module(package_name, pip_command)
-    # get_package_description does not force "develop" branch.
-    return _get_package_description(
-        module_id="netconnectd-daemon",
-        tier=tier,
-        package_name=package_name,
-        displayName="Netconnectd Daemon",
-        displayVersion=version,
-        repo="netconnectd_mrbeam",
-        branch=branch,
-        branch_default=branch,
-        pip="https://github.com/mrbeam/netconnectd_mrbeam/archive/{target_version}.zip",
-        pip_command=pip_command,
-    )
-
-
-def _set_info_iobeam(plugin, tier, beamos_date):
-    if beamos_date > BEAMOS_LEGACY_DATE:
-        pip_command = "sudo /usr/local/iobeam/venv/bin/pip"
-    else:
-        pip_command = GLOBAL_PIP_COMMAND
-    return _get_package_description_with_version(
-        module_id="iobeam",
-        tier=tier,
-        package_name="iobeam",
-        pip_command=pip_command,
-        displayName="iobeam",
-        type="bitbucket_commit",
-        repo="iobeam",
-        api_user="MrBeamDev",
-        api_password="v2T5pFkmdgDqbFBJAqrt",
-        pip="git+ssh://git@bitbucket.org/mrbeam/iobeam.git@{target_version}",
-    )
-
-
-def _set_info_mrb_hw_info(plugin, tier, beamos_date):
-    if beamos_date > BEAMOS_LEGACY_DATE:
-        pip_command = "sudo /usr/local/iobeam/venv/bin/pip"
-    else:
-        pip_command = GLOBAL_PIP_COMMAND
-    return _get_package_description_with_version(
-        module_id="mrb_hw_info",
-        tier=tier,
-        package_name="mrb-hw-info",
-        pip_command=pip_command,
-        displayName="mrb_hw_info",
-        type="bitbucket_commit",
-        repo="mrb_hw_info",
-        api_user="MrBeamDev",
-        api_password="v2T5pFkmdgDqbFBJAqrt",
-        pip="git+ssh://git@bitbucket.org/mrbeam/mrb_hw_info.git@{target_version}",
-    )
-
+def get_sw_update_file_path(plugin):
+    """
+    returns the path to the sw update file
+    @param plugin:
+    @return: path to file
+    """
+    return join(plugin._settings.getBaseFolder("base"), SW_UPDATE_FILE)
 
 @logExceptions
-def _get_octo_plugin_description(module_id, tier, plugin, **kwargs):
-    """Additionally get the version from plugin manager (doesn't it do that by default??)"""
-    # Commented pluginInfo -> If the module is not installed, then it Should be.
-    pluginInfo = plugin._plugin_manager.get_plugin_info(module_id)
-    if pluginInfo is None:
-        display_version = None
+def _set_info_from_file(plugin, tier, beamos_date, _softwareupdate_handler):
+    """
+    loads update info from the update_info.json file
+    the override order: default_settings->module_settings->tier_settings->beamos_settings
+    and if there are update_settings set in the config.yaml they will replace all of the module
+    the json file should look like:
```

Comment:
I think the name is misleading. It sounds like a function that would take a `file` or a `path` variable, and simply parse the content.

I would recommend to have a parser function separately and some other function that takes care of providing that file, use the parser and return the result to the OctoPrint hook.

Reply:
I don't see why i should seperate this, it will only open the file catch some exceptions and the parse the content

Reply:
Please use a(n) `(R)Lock` if you want to prevent simultaneous file access. 

Using singletons is more of a coding style for property managers, which I don't think is necessarily good. They carry a baggage of globalised side effects and their influence on the behaviour of other components is difficult to test because they are often coded to be mutable like the settings manager.

Reply:
for this methode is no singelton used

Reply:
as discussed in the sync meeting, i will use the same lock as for the loading of the file